### DISCUSSION TITLE
Does lldap support roles?

### DISCUSSION THREAD

**Discussion by a137872798** [lldap/lldap#619] [Q&A]
Besides user and group， i need role.

**Comment by nitnelave** [comment]
Can you expand on the notion of role? What do you need it for, what does it represent?

Usually you assign users to groups, and other services grant permissions (or roles) based on the user's groups.

Or if you're talking about permissions for LLDAP, we have the special groups lldap_admin, lldap_readonly and so on.

**Comment by a137872798** [comment]
Emm, maybe i think role is a collect of menu/action permission,and  a group associate a lot of roles. User can join group, now user gain a lot of permissions by group. When i sync user info from ldap. Client can get user info, user group, and user roles.

**Reply by nitnelave** [reply]
Right, but LLDAP doesn't know about the permissions of other services. So if you want to give admin rights to jellyfin, for instance, you would create a jellyfin_admin group in LLDAP, then configure jellyfin to give admin rights to the members of jellyfin_admin.

I don't know what permissions you're trying to give in which service, so it's a bit hard for me to help with the specifics, but that's the idea.

The same thing applies for permissions in LLDAP itself: members of lldap_admin have all admin rights, members of lldap_password_manager can see all users and reset their passwords, and members of lldap_strict_readonly can see all users. There is an association between the group you're in and the permissions you get, similar to a role.

**Reply by a137872798** [reply]
thank you
